contrib/lkt/language/parser.py

```diff
@@ -617,6 +619,12 @@ def called_decl_type():
             lambda _: PropertyError(T.TypeDecl.entity, "Should not happen")
         )
 
+    @langkit_property()
+    def is_generic_fun_decl():
+        return Entity.is_a(T.GenericDecl).then(
+            lambda _: Entity.cast(T.GenericDecl).decl.is_a(T.FunDecl)
```

Comment:
Just using `.cast` should be shorter and clearer: `Entity.cast(T.GenericDecl)._.decl.is_a(T.FunDecl)` Also please add a docstring.